src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -343,6 +343,7 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
   def singleton(tp: Type)(implicit ctx: Context): Tree = tp match {
     case tp: TermRef => ref(tp)
     case tp: ThisType => This(tp.cls)
+    case tp: SkolemType => singleton(tp.narrow)
     case SuperType(qual, _) => singleton(qual)
```

Comment:
I thought `SkolemType` is only used for existential types, but it seems not to be the case. Could you briefly explain a little bit the usage of `SkolemType`, or add a comment to `class SkolemType`?

`RecThis` is also a `SingletonType`, I guess it can be safely ignored here.